tfutils/tests/test.py

```diff
@@ -12,46 +13,135 @@
+    test_multimodel_training()
+    test_distributed_multimodel_training()
+
+
+def run_training_save_tests():
+    """Run training save tests."""
+    test_training_save()
+    test_distributed_training_save()
+    test_multimodel_training_save()
+    test_distributed_multimodel_training_save()
+
+
+def run_custom_training_tests():
+    """Run custom training tests."""
+    test_custom_training()
+    test_custom_distributed_training()
+    test_custom_multimodel_training()
+    test_custom_distributed_multimodel_training()
+
+
+def run_feature_extraction_tests():
+    """Run feature extraction tests."""
+    test_feature_extraction()
+    test_distributed_feature_extraction()
+    test_multimodel_feature_extraction()
+    test_distributed_multimodel_feature_extraction()
+
+
+def run_validation_tests():
+    """Run validation tests."""
+    test_validation()
+    test_distributed_validation()
+    test_multimodel_validation()
+    test_distributed_multimodel_validation()
+
 
 def test_training():
-    """This test illustrates how basic training is performed using the
-       tfutils.base.train_from_params function.  This is the first in a sequence of
-       interconnected tests. It creates a pretrained model that is used by
-       the next few tests (test_validation and test_feature_extraction).
+    """Illustrate training.
+
+    This test illustrates how basic training is performed using the
+    tfutils.base.train_from_params function.  This is the first in a sequence of
+    interconnected tests. It creates a pretrained model that is used by
+    the next few tests (test_validation and test_feature_extraction).
 
-       As can be seen by looking at how the test checks for correctness, after the
-       training is run, results of training, including (intermittently) the full variables
-       needed to re-initialize the tensorflow model, are stored in a MongoDB.
+    As can be seen by looking at how the test checks for correctness, after the
+    training is run, results of training, including (intermittently) the full
+    variables needed to re-initialize the tensorflow model, are stored in a
+    MongoDB.
+
+    Also see docstring of the tfutils.base.train_from_params function for more detailed
+    information about usage.
 
-       Also see docstring of the tfutils.base.train_from_params function for more detailed
-       information about usage.
     """
     # delete old database if it exists
     conn = pm.MongoClient(host=testhost,
                           port=testport)
-    conn.drop_database(testdbname)
-    nm = testdbname + '_' + testcol + '_training0'
-    [conn.drop_database(x) for x in conn.database_names() if x.startswith(nm) and '___RECENT' in x]
-    nm = testdbname + '_' + testcol + '_training1'
-    [conn.drop_database(x) for x in conn.database_names() if x.startswith(nm) and '___RECENT' in x]
+
+    # delete old collection if it exists
+    # conn[testdbname][testcol + '.files'].drop()
```

Comment:
actually yeah let's do a real test suite using nose or nose2 or whatever, and with proper setup/teardown